WebApi/WebApi/Controllers/PostController.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using WebApi.Models.DTO;
+using WebApi.Models.DTO.PostDTOs;
+using WebApi.Services.Services_Interfaces;
+
+
+namespace WebApi.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class PostController : Controller
+    {
+        private readonly IPostService _postService;
+        private readonly ILogger<PostController> _logger;
+        public PostController(ILogger<PostController> logger , IPostService postService)
+        {
+            _logger = logger;
+            _postService = postService;
+        }
+
+
+        [HttpGet]
+        public IQueryable<PostDTO> GetAll([FromHeader] int userID)
+        {
+            return _postService.GetAll();
+        }
+
+        //Proponuje przenieść to do kontrolera user. Zniknie konflikt posts/userID i posts/postID
+        [HttpGet("{userID}")]
+        public IQueryable<PostDTO> GetUserPosts([FromQuery] int UserID) //[FromHeader] int headerID
+        {
+            return _postService.GetAllOfUser(UserID);
+        }
+
+        [HttpGet("{postID}")]
+        public PostDTO Get([FromHeader] int userID, [FromRoute] int id)
+        {
+            return _postService.GetById(id);
+        }
+
+
+        [HttpDelete("{postID}")]
+        public Task Delete([FromHeader] int userID, [FromRoute] int postID, [FromBody] DateTime dateTime)
+        {
+            return _postService.DeletePostAsync(postID);
+        }
+
+        [HttpPut("{postID}")]
+        public Task Edit([FromHeader] int userID, [FromRoute] int postID, [FromBody] PostEditDTO body)
```

Comment:
tu przy zwracaniu ma być po prostu json z success i status code, tak jak w dokumentacji, nwm czy to ma sens najwyżej sie dopytamy

[Required] przed [FromHeader]